### PR TITLE
MAYA-111932 no USD prompt during crash

### PR DESCRIPTION
**Pull request by pierrebai-adsk**
- Detect if the Maya SDK supports the new crash check function.
- If present, use it to determine if Maya is crashing during auto-save.
- If Maya is crashing, avoid asking the user and just save the USD data inside the Maya recovery file.

### CODE REVIEW

File: cmake/modules/FindMaya.cmake [hunk 1]
```diff
@@ -361,6 +361,15 @@ if (MAYA_INCLUDE_DIRS AND EXISTS "${MAYA_INCLUDE_DIR}/maya/MSelectionContext.h")
     endif()
 endif()
 
+set(MAYA_HAS_CRASH_DETECTION FALSE CACHE INTERNAL "isInCrashHandler")
+if(MAYA_INCLUDE_DIRS AND EXISTS "${MAYA_INCLUDE_DIR}/maya/MGlobal.h")
+    file(STRINGS ${MAYA_INCLUDE_DIR}/maya/MGlobal.h MAYA_HAS_API REGEX "isInCrashHandler")
+    if(MAYA_HAS_API)
+        set(MAYA_HAS_CRASH_DETECTION TRUE CACHE INTERNAL "isInCrashHandler")
+        message(STATUS "Maya has isInCrashHandler API")
```
Comment (HamedSabri-adsk) [inline cmake/modules/FindMaya.cmake:369]: Hi @pierrebai-adsk , I am looking at Maya source code and no where can I find "isInCrashHandler"? When was this flag added?

Comment (HamedSabri-adsk) [reply cmake/modules/FindMaya.cmake:369]: @pierrebai-adsk nvm my previous comment. I just saw your Maya PR.  Thanks

File: lib/mayaUsd/nodes/layerManager.cpp [hunk 2]
```diff
@@ -448,7 +458,10 @@ bool LayerDatabase::saveUsd(bool isExport)
     auto opt = MayaUsd::utils::serializeUsdEditsLocationOption();
 
     if (MayaUsd::utils::kIgnoreUSDEdits != opt) {
-        if (_batchSaveDelegate && _proxiesToSave.length() > 0) {
+        if (isCrashing()) {
+            result = kPartiallyCompleted;
+            opt = MayaUsd::utils::kSaveToUSDFiles;
```
Comment (ppt-adsk) [inline lib/mayaUsd/nodes/layerManager.cpp:463]: Your pull request description says you'll save inside the Maya file when crashing, but your code says save to USD files.  Which one should it be?

Comment (pierrebai-adsk) [reply lib/mayaUsd/nodes/layerManager.cpp:463]: Is Maya, thanks for catching this.

**Review by HamedSabri-adsk — APPROVED** [review]
Lgtm!

**Review by ppt-adsk — APPROVED** [review]

**Comment by kxl-adsk** [comment]
@pierrebai-adsk We merged a number of changes. Please resolve merge conflicts and run PF.

**Comment by pierrebai-adsk** [comment]
I'll be honest, I don't understand why clang-format made me reformat MenuBuilder.cpp. It seems to me the line was shorter than 100 columns limit...?

**Pull request merged**